### PR TITLE
add h2c server to echo-basic

### PR DESCRIPTION
**Pull request by dprotaso**
Pulls the image changes from: https://github.com/kubernetes-sigs/gateway-api/pull/2456

/kind test

```release-note
NONE
```

### CODE REVIEW

File: conformance/echo-basic/echo-basic.go [hunk 1]
```diff
@@ -152,6 +161,38 @@ func delayResponse(request *http.Request) error {
...
+		Handler:           h2c.NewHandler(handler, &http2.Server{}),
+	}
+	fmt.Printf("Starting server, listening on port %s (h2c)\n", h2cPort)
+	err := h1s.ListenAndServe()
+	if err != nil {
+		errchan <- err
+	}
+}
+
+func h2cHandler(w http.ResponseWriter, r *http.Request) {
```
Comment (robscott) [inline conformance/echo-basic/echo-basic.go:186]: This doesn't look like it's used anymore?

File: conformance/echo-basic/echo-basic.go [hunk 2]
```diff
@@ -152,6 +161,38 @@ func delayResponse(request *http.Request) error {
...
+	handler := http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		if r.ProtoMajor != 2 {
+			w.WriteHeader(http.StatusBadRequest)
+			fmt.Fprint(w, "Expected h2(c) request")
+			return
+		}
+
+		echoHandler(w, r)
+	})
+	h1s := &http.Server{
```
Comment (robscott) [inline conformance/echo-basic/echo-basic.go:174]: Don't you mean `h2s` here?

**Review by robscott — COMMENTED** [review]
Thanks @dprotaso!

File: conformance/echo-basic/echo-basic.go (hunk 2 above)
Comment (dprotaso) [reply conformance/echo-basic/echo-basic.go:174]: so there's an http2 package in x/net that has a `Server` - so it's technically not an h2c server

Comment (dprotaso) [reply conformance/echo-basic/echo-basic.go:174]: I'll just call it h2c server

**Comment by dprotaso** [comment]
Tested with the following commands

```sh
$ curl  -vvv --http2-prior-knowledge localhost:3001
succeeds

$ curl  -vvv --http2 localhost:3001
succeeds

$ curl  -vvv localhost:3001
fails (as expected)
```

**Comment by robscott** [comment]
Thanks @dprotaso!

/lgtm

**Pull request merged**